src/verilog/program/type_check.cc

```diff
@@ -456,9 +457,24 @@ void TypeCheck::visit(const ContinuousAssign* ca) {
   net_lval_ = false;
   ca->get_assign()->get_rhs()->accept(this);
 
-  const auto r = Resolve().get_resolution(ca->get_assign()->get_lhs());
-  if ((r != nullptr) && (dynamic_cast<const NetDeclaration*>(r->get_parent()) == nullptr)) {
-    error("Continuous assignments are only permitted for variables with type wire", ca);
+  const auto* l = Resolve().get_resolution(ca->get_assign()->get_lhs());
+  if (l != nullptr) {
```

Comment:
Very small nit-pick. The google coding style guidelines (which is more or less what we try to adhere to) say to avoid lots of nested conditionals. Since this conditional guards the remainder of the method, it would be better to change this to:

if (l == nullptr) {
  return;
}

and then have the remainder of your code follow.